### PR TITLE
Added CDF and corrected PDF for GumbelDistribution

### LINKED ISSUE #16521: Incorrect probability density function for Gumbel
The formula for density (pdf) of [Gumbel distribution](https://en.wikipedia.org/wiki/Gumbel_distribution) is incorrect.
```py
In [2]: mu = Symbol("mu")                                                       

In [3]: beta = Symbol("beta", positive=True)                                    

In [4]: X = Gumbel("X", beta, mu) 

In [5]: density(X)(x)                                                           
Out[5]: 
  -(-μ + x)          
  ──────────         
      β        -μ + x
 ℯ           - ──────
                 β   
ℯ                    
─────────────────────
          β          
```
It was introduced in #11751. It should be:
```py
    -(-μ + x)          
    ──────────         
        β        -μ + x
 - ℯ           - ──────
                   β   
ℯ                      
───────────────────────
           β           

```
The [docstring](https://github.com/sympy/sympy/blob/master/sympy/stats/crv_types.py#L1459-L1497) for Gumbel should be improved as well.

Originated from https://github.com/sympy/sympy/pull/16381#issuecomment-478355173.

### PR DESCRIPTION
**Pull request by czgdp1807**
#### References
[1] Fixes https://github.com/sympy/sympy/issues/16521
[2] https://en.wikipedia.org/wiki/Gumbel_distribution

#### Brief description of what is fixed or changed
PDF for GumbelDistribution has been corrected with changes made at appropriate places(docs, tests, code)
CDF for GumbelDistribution has been added. Associated test have also been added.

#### Other comments
Code:
```
from sympy import *
import sympy.stats as ss
beta, mu = symbols('beta, mu', real=True)
x = symbols('x', real=True)
X = ss.Gumbel('X', beta, mu)
print(ss.cdf(X).doit())
```
Outputs:
Prior to this addition:
`Lambda(_z, exp(b/eta)*Integral(exp(-_x/eta)*exp(exp(-_x/eta)*exp(b/eta)), (_x, -oo, _z))/eta)`

With this addition:
`Lambda(_z, exp(-exp(_z - b)/eta))`
#### Release Notes

* stats
  * PDF of Gumbel distribution corrected.
  * CDF of Gumbel distribution added.

### CODE REVIEW

**Comment by czgdp1807** [comment]
@jksuom A trivial addition. Please see if it works.

File: sympy/stats/tests/test_continuous_rv.py [hunk 1]
```diff
@@ -430,6 +430,7 @@ def test_gumbel():
     x = Symbol("x")
     X = Gumbel("x", beta, mu)
     assert simplify(density(X)(x)) == exp((beta*exp((mu - x)/beta) + mu - x)/beta)/beta
+    assert cdf(X)(x) == exp(-exp(-mu + x)/beta)
```
Comment (sidhantnagpal) [inline sympy/stats/tests/test_continuous_rv.py]: This should return `exp(-exp((mu - x)/beta))`.

Comment (czgdp1807) [reply sympy/stats/tests/test_continuous_rv.py]: I have made the corrections. Please check @sidhantnagpal .

**Review by supreet11agrawal — CHANGES_REQUESTED** [review]
https://github.com/sympy/sympy/blob/b475b451b3ad89270baa33a5939a0035a5c5343e/sympy/stats/crv_types.py#L1486
Below this, perhaps `cdf` example can be added.

File: sympy/stats/tests/test_continuous_rv.py [hunk 2]
```diff
@@ -430,6 +430,7 @@ def test_gumbel():
     x = Symbol("x")
     X = Gumbel("x", beta, mu)
     assert simplify(density(X)(x)) == exp((beta*exp((mu - x)/beta) + mu - x)/beta)/beta
+    assert cdf(X)(x) == exp(-exp((-mu + x)/beta))
```
Comment (sidhantnagpal) [inline sympy/stats/tests/test_continuous_rv.py]: The sign is still not correct. It should be `mu - x`, see the formula at https://en.wikipedia.org/wiki/Gumbel_distribution#Properties.

Comment (czgdp1807) [reply sympy/stats/tests/test_continuous_rv.py]: @sidhantnagpal Corrected. Apologies for a lot of mistakes.

**Comment by czgdp1807** [comment]
@sidhantnagpal Any updates/comments on the PR?

File: sympy/stats/crv_types.py [hunk 3]
```diff
@@ -1442,6 +1442,10 @@ def pdf(self, x):
         beta, mu = self.beta, self.mu
         return (1/beta)*exp(-((x-mu)/beta)+exp(-((x-mu)/beta)))
 
+    def cdf(self, x):
```
Comment (sidhantnagpal) [inline sympy/stats/crv_types.py]: The method name `_cdf` is used for the probability distributions.
```suggestion
    def _cdf(self, x):
```

**Comment by czgdp1807** [comment]
@sidhantnagpal Please let me know if there is anything else to change.

**Comment by sidhantnagpal** [comment]
Looks fine to me, can you add an example for the docs?

**Comment by sidhantnagpal** [comment]
It is also worth noting that with appropriate assumptions set, the result currently is the following (which also has an `oo` term):
```
In [2]: mu = Symbol("mu")                                                       

In [3]: beta = Symbol("beta", positive=True)                                    

In [4]: X = Gumbel("x", beta, mu)

In [5]: cdf(X)(x)                                                              
Out[5]: 
⎛           μ  -x     ⎞   
⎜     -μ    ─  ───    ⎟  μ
⎜     ───   β   β     ⎟  ─
⎜      β   ℯ ⋅ℯ       ⎟  β
⎝- β⋅ℯ   ⋅ℯ        + ∞⎠⋅ℯ 
──────────────────────────
            β             
```

The problem is that the formula for density (pdf) of Gumbel is wrong.
It seems it was introduced in https://github.com/sympy/sympy/pull/11751.

With the correct formula for density(X)(x), cdf(X)(x) does give the correct result, however it might be slow, so the changes here (#16381) are OK.

**Comment by czgdp1807** [comment]
@sidhantnagpal Kindly check, the updates. 
I believe that using pre-existing results is useful since, it helps in speeding up the process of generating output. Moreover, the correctness is not affected irrespective of the changes made in modules outside sympy.stats.

File: sympy/stats/crv_types.py [hunk 4]
```diff
@@ -1440,7 +1440,12 @@ class GumbelDistribution(SingleContinuousDistribution):
 
     def pdf(self, x):
         beta, mu = self.beta, self.mu
-        return (1/beta)*exp(-((x-mu)/beta)+exp(-((x-mu)/beta)))
+        z = (x-mu)/beta
+        return (1/beta)*exp(-(z+exp(-z)))
```
Comment (sidhantnagpal) [inline sympy/stats/crv_types.py]: Maybe just `exp(-(z + exp(-z)))/beta`. (Consider adding spaces around binary operators `+` and `-`.)

File: sympy/stats/tests/test_continuous_rv.py [hunk 5]
```diff
@@ -429,7 +429,8 @@ def test_gumbel():
     mu = Symbol("mu")
     x = Symbol("x")
     X = Gumbel("x", beta, mu)
-    assert simplify(density(X)(x)) == exp((beta*exp((mu - x)/beta) + mu - x)/beta)/beta
+    assert simplify(density(X)(x)) == exp(-exp(-(-mu + x)/beta) - (-mu + x)/beta)/beta
```
Comment (sidhantnagpal) [inline sympy/stats/tests/test_continuous_rv.py]: Can we drop the `simplify` here?

Comment (czgdp1807) [reply sympy/stats/tests/test_continuous_rv.py]: Agreed, it just slows down the code. This shouldn't have been there. I don't know the reason placing it there.

File: sympy/stats/crv_types.py (hunk 4 above)
Comment (czgdp1807) [reply sympy/stats/crv_types.py]: I forgot to do that here. Thanks for correcting.

**Comment by sidhantnagpal** [comment]
Let's help the docstring of `Gumbel`, as well. The following should do it:
```diff
@@ -1462,15 +1463,16 @@ def Gumbel(name, beta, mu):
     The density of the Gumbel distribution is given by
 
     .. math::
-        f(x) := \exp \left( -exp \left( x + \exp \left( -x \right) \right) \right)
+        f(x) := \dfrac{1}{\beta} \exp \left( -\dfrac{x-\mu}{\beta}
+                - \exp \left( -\dfrac{x - \mu}{\beta} \right) \right)
 
-    with ::math 'x \in [ - \inf, \inf ]'.
+    with :math:`x \in [ - \infty, \infty ]`.
```

**Comment by czgdp1807** [comment]
@sidhantnagpal Done. Please see if it LGTY.

**Comment by sidhantnagpal** [comment]
Looking good. Can you rebase it onto 1.4 branch and squash the commits with an appropriate message?

File: sympy/stats/tests/test_continuous_rv.py (hunk 5 above)
Comment (czgdp1807) [reply sympy/stats/tests/test_continuous_rv.py]: I got it. It shouldn't be dropped. May be, there are comparison issues. Let's keep it there.

Comment (sidhantnagpal) [reply sympy/stats/tests/test_continuous_rv.py]: The tests could be just:
```diff
--- a/sympy/stats/tests/test_continuous_rv.py
+++ b/sympy/stats/tests/test_continuous_rv.py
@@ -443,7 +443,8 @@ def test_gumbel():
     mu = Symbol("mu")
     x = Symbol("x")
     X = Gumbel("x", beta, mu)
-    assert simplify(density(X)(x)) == exp((beta*exp((mu - x)/beta) + mu - x)/beta)/beta
+    assert str(density(X)(x)) == 'exp(-exp(-(-mu + x)/beta) - (-mu + x)/beta)/beta'
+    assert cdf(X)(x) == exp(-exp((mu - x)/beta))
```

**Pull request closed without merging**